src/components/Contacts/ContactFilters/ContactFilters.tsx

```diff
@@ -1,35 +1,211 @@
+import { ArrowBackIos, ArrowForwardIos } from '@material-ui/icons';
+import { isArray } from 'lodash';
+import React, { useState } from 'react';
+import { useTranslation } from 'react-i18next';
+import { ContactFilterGroup } from '../../../../graphql/types.generated';
+import { Filter } from '../../Shared/Filters/Filter';
+import { FilterListItem } from '../../Shared/Filters/FilterListItem';
+import { FilterListItemShowAll } from '../../Shared/Filters/FilterListItemShowAll';
+import { useContactFiltersQuery } from './ContactFilters.generated';
+
+const FilterHeader = styled(Box)(({ theme }) => ({
+  padding: theme.spacing(2),
+  borderBottom: '1px solid',
+  borderBottomColor: theme.palette.grey[200],
+}));
+
+const FilterList = styled(List)(({ theme }) => ({
+  '& .MuiListItemIcon-root': {
+    minWidth: '37px',
+  },
+  '& .FilterListItemMultiselect-root': {
+    marginBottom: theme.spacing(4),
+  },
+}));
+
+const LinkButton = styled(Button)(() => ({
+  minWidth: 0,
+  textTransform: 'none',
+}));
 
 interface Props {
   accountListId: string;
 }
 
-export const ContactFilters: React.FC<Props> = ({ accountListId }: Props) => {
-  const [
-    loadContactFilters,
-    { data, loading, error },
-  ] = useContactFiltersLazyQuery({
+export const ContactFilters: React.FC<Props & BoxProps> = ({
+  accountListId,
+  ...boxProps
+}) => {
+  const theme = useTheme();
+  const { t } = useTranslation();
+
+  const { data, loading, error } = useContactFiltersQuery({
     variables: { accountListId },
   });
 
+  const [selectedGroup, showGroup] = useState<null | ContactFilterGroup>(null);
+  const [selectedFilters, setSelectedFilters] = useState({});
+  const [showAll, setShowAll] = useState(false);
+
+  const updateSelectedFilter = (name: string, value) => {
+    if (value && !(isArray(value) && value.length == 0))
+      setSelectedFilters((prev) => {
+        return { ...prev, [name]: value };
+      });
+    else
+      setSelectedFilters((prev) => {
+        const updated = { ...prev };
+        delete updated[name];
+        return updated;
+      });
+  };
+
+  const getSelectedFilters = (group: ContactFilterGroup) =>
+    group.filters.filter((value) => selectedFilters[value.name]);
+  const isGroupVisible = (group: ContactFilterGroup) =>
+    group.alwaysVisible || getSelectedFilters(group).length > 0;
+
   return (
-    <div style={{ backgroundColor: colors.amber[600] }}>
-      <h2>Filters</h2>
-      <button onClick={() => loadContactFilters()}>Load Filters</button>
-      {error && <p data-testID="ErrorText">Error: {error.toString()}</p>}
-      {loading ? (
-        <p>Loading Filters</p>
-      ) : !data || data.contactFilters.length === 0 ? (
-        <p>No Filters</p>
-      ) : (
-        <ul data-testID="FiltersList">
-          {data.contactFilters.map(({ id, name }) => (
-            <li key={id}>{name}</li>
-          ))}
-        </ul>
-      )}
-    </div>
+    <Box {...boxProps}>
+      <div style={{ overflow: 'hidden' }}>
```

Comment:
this `overflow` attribute is to isolate the horizontal slide animation within the filters div. otherwise the UI slides across the entire screen.

Honestly the current horizontal slide animation needs work, it has positioning issues while it's animating. it goes fast enough that you may not notice it unless you are specifically looking for it